### DISCUSSION TITLE
Converting simple .DOCX after long time fails

### DISCUSSION THREAD

**Discussion by jannis6023** [gotenberg/gotenberg#585] [Q&A]
Hi,
I am just implementing Gotenberg.

The first request after a long time (12h+ - don't know if it would be after a shorter time) fails, axios logs `socket hang up` and the Container logs the following:
`
{"level":"error","ts":1673542663.6736646,"logger":"api","msg":"convert to PDF: unoconv PDF: context done: context deadline exceeded","trace":"e520ee81-0a90-42cf-a90c-b584aaaad62a","remote_ip":"172.19.0.1","host":"192.168.21.2:3002","uri":"/forms/libreoffice/convert","method":"POST","path":"/forms/libreoffice/convert","referer":"","user_agent":"axios/1.2.1","status":503,"latency":30001182640,"latency_human":"30.00118264s","bytes_in":-1,"bytes_out":19}
`

What might the problem be here?

Thanks in advance,
Jannis

**Comment by gulien** [comment]
Hello @jannis6023,

You should increase the API timeout with:

```
--api-timeout duration  Set the time limit for requests (default 30s)
```

For instance, `--api-timeout=60s`

**Comment by jannis6023** [comment]
EDIT: I just read an other issue concerning some error like this... but why is it taking this long ONLY on the first conversion? If I try it just a second later, everything's working on the SAME FILE in a second!

**Comment by jannis6023** [comment]
> Hello @jannis6023,
> 
> You should increase the API timeout with:
> 
> ```
> --api-timeout duration  Set the time limit for requests (default 30s)
> ```
> 
> For instance, `--api-timeout=60s`

thanks, I'll try this. but: is there any cause for the long delay on the first request?
What do I have to put in when using docker compose trying to implement your response?

**Comment by gulien** [comment]
> but: is there any cause for the long delay on the first request?

Can't remenber, but "big" Word file may be very long to convert. Also 12h (!) long seems to be an issue on the client side (afaik), because Gotenberg correctly returns a `503` response after 30 seconds.

> What do I have to put in when using docker compose trying to implement your response?

See https://gotenberg.dev/docs/get-started/docker-compose#modules-properties.
